libs/ast_generic/AST_generic.ml

```diff
@@ -1991,8 +1998,21 @@ let basic_id_info ?(hidden = false) resolved =
 
 (* TODO: move AST_generic_helpers.name_of_id and ids here *)
 
-let dotted_to_canonical xs = Common.map fst xs
-let canonical_to_dotted tid xs = xs |> Common.map (fun s -> (s, tid))
+let canonical_append { unqualified; _ } { unqualified = unqualified2; tok } =
+  { unqualified = unqualified @ unqualified2; tok }
+
+let dotted_to_canonical xs =
```

Comment:
Could we avoid this function? I only see it being used together with `canonical_append`, so why not combine them both?

Reply:
I think I used it a fair few times, such as in `Naming_AST`, even without `canonical_append`. Basically, we need to produce a `canonical_name` either way to fit into the existing framework (since that's what `ImportedEntity` and co take in), so I don't really see a way of avoiding this function.

Reply:
Oh, I see what happened. There's a few times it was already used (the function being pre-existing prior to this PR), so they won't show up in the diff. But there are like 10 occurrences of it in `Naming_AST`.